src/main/java/org/generationcp/middleware/dao/dms/StockDao.java

```diff
@@ -335,7 +350,24 @@ public void replaceExperimentStocks(final Integer oldStockId, final Integer newS
 			LOG.error(errorMessage, e);
 			throw new MiddlewareQueryException(errorMessage, e);
 		}
+	}
 
+	public Integer getNextEntryNumber(final Integer studyId) {
+		try {
+			final String queryString = "SELECT MAX(Convert(s.uniquename, SIGNED)) FROM stock s where s.project_id = :studyId";
```

Comment:
@mderamos would not be clear to use IFNULL in the query instead og having a IF in java? also you can sum + 1 in the query
something like 
`SELECT IFNULL(MAX(Convert(s.uniquename, SIGNED)), 0) + 1 FROM stock s where s.project_id = :studyId`